### PR TITLE
Compute payout refactor

### PR DESCRIPTION
**Pull request by Eknir**
See comments [8dab06](https://github.com/ethersphere/swap-swear-and-swindle/commit/08dab063f0b5b4aa01f7da95014557763888ee50).

### CODE REVIEW

File: contracts/SimpleSwap.sol [hunk 1]
```diff
@@ -8,9 +8,9 @@ contract SimpleSwap {
   using SafeMath for uint;
 
   event Deposit(address depositor, uint amount);
-  event ChequeCashed(address indexed beneficiary, uint indexed serial, uint amount);
+  event ChequeCashed(address indexed beneficiary, uint indexed serial, uint payout, uint requestPayout);
   event ChequeSubmitted(address indexed beneficiary, uint indexed serial, uint amount, uint timeout);
-  event ChequeBounced(address indexed beneficiary, uint indexed serial, uint paid, uint bounced);
+  event ChequeBounced();
```
Comment (diegomasini) [inline contracts/SimpleSwap.sol]: Does it make sense to keep this event?

Comment (ralph-pichler) [reply contracts/SimpleSwap.sol]: yes, it is an easy way to see if a swap channel was not solved at some time.

File: contracts/SimpleSwap.sol [hunk 2]
```diff
@@ -121,58 +121,35 @@ contract SimpleSwap {
...
-
+    require(now >= cheque.timeout,  "SimpleSwap: cheque not yet timed out");
+    require(requestPayout <= cheque.amount.sub(cheque.paidOut), "SimpleSwap: not enough balance owed");
+    /* ensure there is a balance to claim */
+     /* calculates hard-deposit usage */
+    uint hardDepositUsage = Math.min(requestPayout, hardDeposits[beneficiary].amount);
+    /* calculates acutal payout */
+    uint payout = Math.min(requestPayout, liquidBalance() + hardDepositUsage);
+    /* will throw if chequebook is not solvent */
+    require(payout != 0, "SimpleSwap: contract not solvent");
```
Comment (ralph-pichler) [inline contracts/SimpleSwap.sol]: I still think this check shouldn't be here. We want a bounced event if the contract is insolvent.

File: contracts/SimpleSwap.sol [hunk 3]
```diff
@@ -121,58 +121,35 @@ contract SimpleSwap {
...
-    }
-  }
-
   /// @notice attempt to cash latest cheque
   /// @param beneficiary beneficiary for whose cheque should be paid out
-  function cashCheque(address payable beneficiary) public {
-    ChequeInfo storage info = cheques[beneficiary];
-
+  /// @param requestPayout amount requested to pay out
+  function cashCheque(address payable beneficiary, uint requestPayout) public returns (uint) {
```
Comment (ralph-pichler) [inline contracts/SimpleSwap.sol]: What's the purpose of the return value? Do we expect some beneficiaries to be contracts?

File: contracts/SimpleSwap.sol (hunk 2 above)
Comment (Eknir) [reply contracts/SimpleSwap.sol]: Agreed, I will delete

File: contracts/SimpleSwap.sol (hunk 3 above)
Comment (Eknir) [reply contracts/SimpleSwap.sol]: I thought I could use it in Swap.sol, but this is not the case. Can be removed for now

Comment (Eknir) [reply contracts/SimpleSwap.sol]: I foresee that we have to add it back later, but for now we can omit it.

**Comment by Eknir** [comment]
@ralph-pichler , I incorporated your review. Something did not go really as planned with the branches and committing, but I think I can't undo this anymore. 
Please let me know if you any further comments related to this PR.

**Pull request merged**